helix-view/src/document.rs

```diff
@@ -115,6 +116,58 @@ pub struct SavePoint {
     revert: Mutex<Transaction>,
 }
 
+// An enum representing the encoding and if it contain a byte order mark
+#[derive(Debug, Clone)]
+pub enum EncodingBom {
+    UTF8,
+    NoBom(&'static Encoding),
+}
```

Comment:
This could be `Copy`. But it also seems a bit of a footgun that it can either be `UTF8` or `NoBom(UTF8)`.

Reply:
I'd rather structure the encoding field on the document to be a tuple `(Encoding, bool)` or a separate bool field. Then we can pattern match:

```rust
match {
  (UTF8, true) =>
  (UTF16_BE, true) =>
  (UTF16_LE, true) =>
  (encoding, _ignore) => 
```

or

```rust
fn has_bom() -> bool {
   matches!((encoding, bom), (UTF8 | UTF16_BE | UTF_16_LE, true))
}
```

Reply:
Funnily enough, I came to the opposite conclusion when exploring what could be done for bom support, in this branch: https://github.com/helix-editor/helix/compare/master...Alexis-Lapierre:helix:wip_conserve_bom_on_file_save

I first created a boolean to track the bom status, but I decided against it because sometime we could enter an invalid state. What should we do if Document is in state `encoding: windows-1252, has_bom: true` ?

I came to the conclusion that having an enum that track the bom status would not allow to be in an incoherent state, and that's why I made this choice.

Reply:
EncodingBom now derive Copy.

Reply:
I replaced EncodingBom by a boolean in [0ea7235](https://github.com/helix-editor/helix/pull/6497/commits/0ea7235f029c5b5b3562f68642db85a9f9344e46), as requested.